### PR TITLE
[WFLY-19286] Add ServerSetupTasks to reload server to a desired stabi…

### PR DESCRIPTION
**Pull request by kabir**
…lity level

https://issues.redhat.com/browse/WFLY-19286

This needs the Arquillian upgrade in  https://github.com/wildfly/wildfly/pull/17832 merged before the tests can pass
____

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

____

### CODE REVIEW

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (jamezp) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:95]: Should this be a final method with a `doTearDown()` to ensure the snapshot is always restored?

**Comment by kabir** [comment]
@jamezp fixed (sorry for the delay)

File: testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/stability/AbstractStabilityServerSetupTaskTest.java
Comment (bstansberry) [inline testsuite/integration/smoke/src/test/java/org/jboss/as/test/smoke/stability/AbstractStabilityServerSetupTaskTest.java:22]: Static imports first.

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:33]: Static imports first.

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:24]: java and javax after statics but before others

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:41]: and ..... ?

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:60]: This is not a valid statement -- two vars but only one expression in the format string

File: testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java
Comment (bstansberry) [inline testsuite/shared/src/main/java/org/wildfly/test/stabilitylevel/StabilityServerSetupSnapshotRestoreTasks.java:201]: s/Reload/take-snapshot

**Review by bstansberry — CHANGES_REQUESTED** [review]
@kabir In general this LGTM to me but...

1) I have some minor comments
2) Please rebase as this branch now has two unrelated commits that have equivalents that are already in main.

Please feel free to merge this when ready; i.e. don't wait for my formal approval of the above as I'll not be around for a while.

**Pull request merged**
